haddock-api/src/Haddock/Interface/LexParseRn.hs

```diff
@@ -164,9 +164,10 @@ outOfScope dflags x =
     Exact name -> warnAndMonospace name  -- Shouldn't happen since x is out of scope
   where
     warnAndMonospace a = do
-      tell ["Warning: '" ++ showPpr dflags a ++ "' is out of scope.\n" ++
-            "    If you qualify the identifier, haddock can try to link it\n" ++
-            "    it anyway."]
+      tell $ fmap (rdrNameErrMsg x)
+        ["Warning: '" ++ showPpr dflags a ++ "' is out of scope.\n" ++
```

Comment:
I'd get rid of `rdrNameErrMsg` here - it is always going to `noLoc` (as the comment on the `Exact` case indicates).

As before, the `"Warning: ` prefix in the error message is redundant.